### ISSUE TITLE
Preserve key order of pipeline configs

### ISSUE THREAD

**Issue by kinghuang** [dagster-io/dagster#2702]
I've created some pipeline preset configs. It looks like Dagster or Dagit is sorting the keys when they're displayed in Dagit's Playground. The original preset config was written with solids and other things in the order of the pipeline's flow. The sorted view is really hard to follow for long configs.

I would prefer if Dagster/Dagit didn't sort preset configs, and presented them in the way they were originally written.

**Example preset file:**

```
solids:
  refine_text:
    inputs:
      raw:
        read:
          parquet:
            path: /tables/supplier_catalog
            index: VENDOR_SERVICE_ID
        lower_cols: true
        sample: 0.01
      profiles:
        read:
          parquet:
            path: /profiles/supplier_catalog
    outputs:
      - refined:
          to:
            csv:
              path: /refined_csv/*.csv
    config:
      input_col: item_description
      supplier_col: l_company_id
  extract_info:
    outputs:
      - info:
          to:
            json:
              path: /info_json/
              orient: records
              lines: true
resources:
  dask:
    config:
      cluster:
        local: {}
```

**As shown in Dagit:**

The keys at every level have been sorted alphabetically.

![image](https://user-images.githubusercontent.com/27340/87179156-824c9d80-c29b-11ea-9146-c550a9c34cb5.png)

**Comment by schrockn** [comment]
Defintely! Just ran into this myself and it is quite annoying :-) Thanks for the report

**Comment by hellendag** [comment]
Looks like this is an issue with loading yaml in python and losing key order at that point, rather than a Dagit/UI issue -- gonna recategorize this issue.

**Comment by tinkerware** [comment]
We've run into this with a preset, and Dagit seems to sort keys lexicographically, while running the same pipeline and preset with `execute_pipeline` works fine. In our case, processing is key-order dependent (not a good choice on our part in the first place, I admit) so this breaks our runs in Dagit.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one can be closed.

The reproducer is built around pipeline preset configs (`solids:` top-level key, presets edited in Dagit Playground). Both of those concepts are gone:

- `PresetDefinition` was removed during the 1.0 transition — the only mention left in the repo is in `MIGRATION.md`: https://github.com/dagster-io/dagster/blob/master/MIGRATION.md
- `solid` / `pipeline` decorators and the `solids:` config namespace were retired in favor of `op` / `job` and `ops:`. The `dagster/__init__.py` no longer exports `solid` or `pipeline`: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/__init__.py
- The Playground UI itself was reworked around the webserver (`dagster-webserver`) and the launchpad pages — the original Dagit Playground component this issue refers to no longer exists.

If config-key ordering is still a pain point in the current launchpad with run config JSON/YAML, opening a fresh issue against the current UI with a screenshot would be much easier to act on. Otherwise this seems safe to close.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `master` branch.)

**Comment by kinghuang** [comment]
Wow, blast from the past! I agree that Dagster's evolved quite a bit since I originally filed this issue. I'll close this one.
